ORStools/proc/directions_lines_proc.py

```diff
@@ -27,163 +27,73 @@
  ***************************************************************************/
 """
 
-import os.path
 from qgis.core import (QgsWkbTypes,
                        QgsCoordinateReferenceSystem,
                        QgsProcessing,
-                       QgsProcessingAlgorithm,
                        QgsProcessingParameterField,
                        QgsProcessingParameterFeatureSource,
                        QgsProcessingParameterEnum,
-                       QgsProcessingParameterFeatureSink,
                        QgsProcessingParameterBoolean,
                        QgsPointXY,
                        )
 
-from PyQt5.QtGui import QIcon
+from ORStools.common import directions_core, PROFILES, PREFERENCES
+from ORStools.utils import transform, exceptions, logger
+from .base_processing_algorithm import ORSBaseProcessingAlgorithm
+from ..utils.processing import get_params_optimize
 
-from ORStools import RESOURCE_PREFIX, __help__
-from ORStools.common import client, directions_core, PROFILES, PREFERENCES
-from ORStools.utils import configmanager, transform, exceptions, logger
-from . import HELP_DIR
 
-
-class ORSdirectionsLinesAlgo(QgsProcessingAlgorithm):
+# noinspection PyPep8Naming
+class ORSDirectionsLinesAlgorithm(ORSBaseProcessingAlgorithm):
     """Algorithm class for Directions Lines."""
-
-    ALGO_NAME = 'directions_from_polylines_layer'
-    ALGO_NAME_LIST = ALGO_NAME.split('_')
-
-    IN_PROVIDER = "INPUT_PROVIDER"
-    IN_LINES = "INPUT_LINE_LAYER"
-    IN_FIELD = "INPUT_LAYER_FIELD"
-    IN_PROFILE = "INPUT_PROFILE"
-    IN_PREFERENCE = "INPUT_PREFERENCE"
-    IN_OPTIMIZE = "INPUT_OPTIMIZE"
-    IN_MODE = "INPUT_MODE"
-    OUT = 'OUTPUT'
-
-    # noinspection PyUnusedLocal
-    def initAlgorithm(self, configuration):
-
-        providers = [provider['name'] for provider in configmanager.read_config()['providers']]
-        self.addParameter(
-            QgsProcessingParameterEnum(
-                self.IN_PROVIDER,
-                "Provider",
-                providers,
-                defaultValue=providers[0]
-            )
-        )
-
-        self.addParameter(
+    def __init__(self):
+        super().__init__()
+        self.ALGO_NAME = 'directions_from_polylines_layer'
```

Comment:
algorithm names could be better as well, but should be left for now, as changing would break modeler builds and scripts using the current algorithms. But we should maybe deprecate and introduce as new names in the future

Reply:
I'll open an issue regarding that.